lambeq/ansatz/circuit.py

```diff
@@ -143,3 +144,109 @@ def _ar(self, box: Box) -> Circuit:
         else:
             circuit >>= Id(cod) @ Bra(*[0]*(dom - cod))
         return circuit
+
+
+def _sim_ansatz_factory(sim_circuit_n=14):
+    """Generate a SimAnsatz class for the specified Sim ansatz.
+
+    Parameters
+    ----------
+    sim_circuit_n : int
+        Circuit number, according to arXiv:1905.10876
+
+    """
+
+    # Map from Sim circuit number to a function specifying
+    # number of params, given the number of qubits
+    param_count_fns = {14: lambda n_qbs: 4 * n_qbs,
+                       15: lambda n_qbs: 2 * n_qbs}
+
+    sim_discopy_classes = {14: Sim14, 15: Sim15}
+
+    if sim_circuit_n not in sim_discopy_classes:
+        raise ValueError(f'Invalid Sim circuit name: {sim_circuit_n}')
+
+    class SimAnsatz(CircuitAnsatz):
```

Comment:
I think it would be better if we put this in the global namespace.

Reply:
Currently I've put the `SimAnsatz` class inside the factory method since it uses the `sim_circuit_n` parameter and the circuit-specific mappings. It doesn't make sense to instantiate a generic SimAnsatz without these.
Another option would be to add `sim_circuit_n` as a constructor  parameter. I prefer the class within the factory method since this way the signatures for the Sim and IQP ansatze are the same.

Reply:
> Another option would be to add sim_circuit_n as a constructor parameter.

Yes I was referring to this. However you have a point, we can keep it as it is I guess.